pkg/collector/corechecks/containers/kubelet/provider/cadvisor/provider_test.go

```diff
@@ -227,7 +227,7 @@ func (suite *ProviderTestSuite) TestPrometheusFiltering() {
 				suite.T().Fatalf("error created kubelet mock: %v", err)
 			}
 
-			prometheus.ParseMetricsWithFilterFunc = func(data []byte, filter []string) ([]*prom.MetricFamily, error) {
+			prometheus.ParseMetricsWithFilterFunc = func(data []byte, filter []string) ([]prom.MetricFamily, error) {
```

Comment:
are you sure you don't want to carry the pointer and instead copy the values ? this could be quiet big no ?

Reply:
See my detailed explanation in the PR comment - the struct is only 56 bytes of headers (not the actual data), and values provide better allocation efficiency, cache locality, and GC behavior.